src/modelmanager.cpp

```diff
@@ -903,4 +903,20 @@ const CustomNodeLibraryManager& ModelManager::getCustomNodeLibraryManager() cons
     return *customNodeLibraryManager;
 }
 
+std::string ModelManager::getPipelinesStatusesAsJson() {
+    std::string response = "{";
+    std::map<std::string, PipelineDefinitionStatus> pipelinesStatuses;
+    pipelineFactory.getPipelinesStatuses(pipelinesStatuses);
+    if (!pipelinesStatuses.empty()) {
+        response += "\n";
+    }
+    for (auto i = pipelinesStatuses.begin(); i != pipelinesStatuses.end(); i++) {
+        response += " \"" + i->first + "\" : \"" + pipelineDefinitionStateCodeToString(i->second.getStateCode()) + "\"";
```

Comment:
Minor:
using += is better than + with std::string and this is happening multiple times in this line.
Alternatively if you want to keep readability you can use std::stringstream

Reply:
And before concatenation you could call std::string.reserve() with roughly estimated size based on pipelineStatuses length